### DISCUSSION TITLE
Unsupported Protocol for WPA2 Enterprise

### DISCUSSION THREAD

**Discussion by xXPerditorXx** [openssl/openssl#22642] [Q&A]
My Problem is, that the current openSSL version isn't compatible with TLSv1.x.
WPA2 Enterprise WpaSupplicant showed me the Error unsupported protocol TLSv3

I'm on Achlinux and the only way to fix TLS was downgrading to OpenSSL 3.0.9-1
(Don't do that, since it will break your dependency's making it hard to undo again.)

**Comment by mattcaswell** [comment]
> My Problem is, that the current openSSL version isn't compatible with TLSv1.x.

This is not correct. All currently supported versions of OpenSSL support TLSv1.x. However you may need to adjust the security level to get TLSv1.1 and below to work. TLSv1.2 should be fine.

> WPA2 Enterprise WpaSupplicant showed me the Error unsupported protocol TLSv3

That error makes no sense since there is no such thing as TLSv3. Is that a copy & paste error? Did you mean TLSv1.3?

Either way I would suggest addressing your issues to the wpa_supplicant project in the first instance. According to the ChangeLog OpenSSL 3.0 support was added in wpa_supplicant 2.10:

https://w1.fi/cgit/hostap/plain/wpa_supplicant/ChangeLog

**Reply by xXPerditorXx** [reply]
Well, I tried as many said in different forums,

 Adding `openssl_conf = openssl_init` to the first line and at the end of the file
```
[openssl_init]
ssl_conf = ssl_sect

[ssl_sect]
system_default = system_default_sect

[system_default_sect]
CipherString = DEFAULT@SECLEVEL=0
```
Nothing of this helped at all. Seemed like the config got ignored completly.

Oh and wpa_supplicant gave me this, but sadly I can't reproduce it, because I'm not at school and downgraded:
`OpenSSL: openssl_handshake -  SSL connect error:xxxxx:SSL routines:unsupported protocol`

**Reply by xXPerditorXx** [reply]
Is there any update on this?

**Reply by xXPerditorXx** [reply]
Is it working?
I don't want to update just yet in case it isn't working...

**Reply by xXPerditorXx** [reply]
I tried it again and I'm still unable to connect to the network.

**Comment by xXPerditorXx** [comment]
```
Mar 14 08:05:54  wpa_supplicant[557]: wlan0: CTRL-EVENT-EAP-PROPOSED-METHOD vendor=0 method=25
Mar 14 08:05:54  wpa_supplicant[557]: wlan0: CTRL-EVENT-EAP-METHOD EAP vendor 0 method 25 (PEAP) selected
Mar 14 08:05:55  wpa_supplicant[557]: SSL: SSL3 alert: write (local SSL3 detected an error):fatal:protocol version
Mar 14 08:05:55  wpa_supplicant[557]: OpenSSL: openssl_handshake - SSL_connect error:0A000102:SSL routines::unsupported protocol
Mar 14 08:05:56  wpa_supplicant[557]: wlan0: CTRL-EVENT-EAP-FAILURE EAP authentication failed
Mar 14 08:05:58  wpa_supplicant[557]: wlan0: Authentication with 32:5a:4c:8d:eb:51 timed out.
Mar 14 08:05:58  wpa_supplicant[557]: wlan0: CTRL-EVENT-DISCONNECTED bssid=32:5a:4c:8d:eb:51 reason=3 locally_generated=1
Mar 14 08:05:58  wpa_supplicant[557]: wlan0: CTRL-EVENT-SSID-TEMP-DISABLED id=0 ssid="GTS-radius" auth_failures=1 duration=10 reason=AUTH_FAILED
Mar 14 08:05:58  wpa_supplicant[557]: BSSID 32:5a:4c:8d:eb:51 ignore list count incremented to 2, ignoring for 10 seconds
Mar 14 08:05:58  wpa_supplicant[557]: wlan0: CTRL-EVENT-DSCP-POLICY clear_all
```

**Reply by t8m** [reply]
You might also need the legacy provider loaded.

**Reply by xXPerditorXx** [reply]
How do I do that?

**Reply by t8m** [reply]
Please see https://github.com/openssl/openssl/blob/master/README-PROVIDERS.md
